src/var_equality.jl

```diff
@@ -60,7 +60,7 @@ end
 function anova(scores::AbstractVector{<:Real}...)
     Nᵢ = [length(g) for g in scores]
     Z̄ᵢ = mean.(scores)
-    Z̄ = mean(Z̄ᵢ)
+    Z̄ = sum(Iterators.flatten(scores))/sum(Nᵢ)
```

Comment:
Alternatively, one could use
```suggestion
    Z̄ = dot(Z̄ᵢ, Nᵢ) / sum(Nᵢ)
```
In a quick benchmark this seemed to be similarly fast, and usually even marginally faster.

Reply:
You'd probably need a very large dataset to see the difference.

Reply:
Actually a tiny dataset is sufficient. Of course, the difference is very small but it seemed to be consistent:
```julia

julia> using Statistics, LinearAlgebra, BenchmarkTools

julia> function f(scores::AbstractVector{<:Real}...)
           Nᵢ = [length(g) for g in scores]
           Z̄ᵢ = mean.(scores)
           Z̄ = sum(Iterators.flatten(scores)) / sum(Nᵢ)
           return Nᵢ, Z̄ᵢ, Z̄
       end
f (generic function with 1 method)

julia> function g(scores::AbstractVector{<:Real}...)
           Nᵢ = [length(g) for g in scores]
           Z̄ᵢ = mean.(scores)
           Z̄ = dot(Z̄ᵢ, Nᵢ) / sum(Nᵢ)
           return Nᵢ, Z̄ᵢ, Z̄
       end
g (generic function with 1 method)

julia> scores = map(n -> rand(n), (3, 9, 12));

julia> @btime f($(scores...));
  33.893 ns (1 allocation: 64 bytes)

julia> @btime g($(scores...));
  33.687 ns (1 allocation: 64 bytes)

julia> scores = map(n -> rand(n), (3, 9, 12, 134));

julia> @btime f($(scores...));
  33.944 ns (1 allocation: 64 bytes)

julia> @btime g($(scores...));
  33.681 ns (1 allocation: 64 bytes)

julia> scores = map(n -> rand(n), (3, 9, 12, 134, 12, 4134, 1231, 122, 12, 1, 23, 58));

julia> @btime f($(scores...));
  34.070 ns (1 allocation: 64 bytes)

julia> @btime g($(scores...));
  33.781 ns (1 allocation: 64 bytes)
```

Reply:
It looks like something else from Z evaluation dominates in the function. But nanoseconds 😏.